main/src/ca/uwaterloo/flix/language/phase/unification/set/Equation.scala

```diff
@@ -17,31 +17,47 @@
 package ca.uwaterloo.flix.language.phase.unification.set
 
 import ca.uwaterloo.flix.language.ast.SourceLocation
+import ca.uwaterloo.flix.language.phase.unification.set.SetFormula.*
 
 import scala.annotation.nowarn
 
 /**
   * Represents an equality equation between the formulas `f1` and `f2` (`f1 ~ f2`).
   *
+  * Equations can be marked as [[Equation.Status.Unsolvable]] during unification, in which case its
+  * written `f1 !~ f2`.
+  *
   * The `@nowarn` annotation is required for Scala 3 compatibility, since the derived `copy` method
   * is private in Scala 3 due to the private constructor. In Scala 2 the `copy` method is still
   * public. However, we do not use the `copy` method anywhere for [[Equation]], so this is fine.
   */
 @nowarn
-case class Equation private(f1: SetFormula, f2: SetFormula, loc: SourceLocation) {
+case class Equation private(f1: SetFormula, f2: SetFormula, status: Equation.Status, loc: SourceLocation) {
 
   /** Returns the sum of the sizes of the formulas in `this`. */
   final def size: Int = f1.size + f2.size
 
   /** Returns a human-readable string of `this`. */
   override final def toString: String = s"$f1 ~ $f2"
```

Comment:
I can see arguments both for and against including the status here.

Reply:
I think that `f1 !~ f2` would be read as an inequality constraint to new people. We could maybe have `f1 ~(error) f2` and `f1 ~(timeout) f2` to not hide the information? or `f1 ~err f2` and `f1 ~to f2`